spark2/src/main/java/org/apache/iceberg/spark/SparkFilters.java

```diff
@@ -177,4 +179,13 @@ private static Object convertLiteral(Object value) {
     }
     return value;
   }
+
+  private static Expression handleEqual(String attribute, Object value) {
+    Object literal = convertLiteral(value);
```

Comment:
This should be moved into the `else` block because literal should not allow creating a `NaN` literal.